WordPress/gutenberg
# Display the DocumentBar for Style Book and Style Revisions

## What?

When viewing Style Book or Style Revisions in the site editor, displays the DocumentBar in header rather than just the title.

Fixes https://github.com/WordPress/gutenberg/issues/51667 and https://github.com/WordPress/gutenberg/issues/54514

## Why?

Makes the header more consistent. Opening the Style Book or Style Revisions overlay now only changes the title and icon, rather than removing the DocumentBar component and replacing it with a string.

## How?

Given the [plans for refactoring and/or removing the EditorCanvasContainer component](https://github.com/WordPress/gutenberg/issues/62216), this PR attempts to make the minimal changes needed to solve the issue

- Adds `title` and `icon` props to `DocumentBar` so it can display them when an overlay is visible
- Updates the `Header` component to pass `title` and `icon` props to `DocumentBar`, rather than displaying the `title` prop in place of `DocumentBar`
- Temporarily removes the command context in the site editor when Style Book and/or Style Revisions are open, clearing the suggested commands that are specific to editing the current entity (e.g. when editing a pattern or a page)

## Testing Instructions

- Go to the site editor and open the global styles sidebar
- Click on the Style Book and Style Revisions icons at the top of the sidebar
- See that the title and icon in the header change when opening those overlays
- See that you can click on the DocumentBar and open the command menu when the overlays are open

### Testing Instructions for Keyboard

No relevant changes

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/1699996/7083e36a-a726-471c-a464-fe4fb5673936
